### ISSUE TITLE
arXiv-2019/03-SCIBERT: Pretrained Contextualized Embeddings for Scientific Text

### ISSUE THREAD

**Issue by BrambleXu** [BrambleXu/knowledge-graph-learning#86]
**一句话总结：**

基于BERT，在scientific text上训练得到的特定领域的预训练语料。

**资源：**

- [pdf](<https://arxiv.org/pdf/1903.10676v1.pdf>)
- [code](<https://github.com/allenai/scibert>)

**关键字：**

- dataset: 
- BERT

**笔记：**

ELMO和BERT这样的模型，can read a sentence and return a **contextualized embedding** for each token which can be used in task-specific neural architectures。基于情景这一点是这些模型厉害的地方。

现在ELMO和BERT公开的pre-train结果都是基于general domain corpora的。

**BERT**：multilayer bidirectional Transformerm. BERT is pretrained on two tasks:
predicting randomly masked tokens and predicting whether two sentences follow each other

**Vocabulary**: BERT uses WordPiece for unsupervised tokenization of the input text. WordPiece,
similar to byte pair encoding, relies on a vocabulary of subword units. The vocabulary is built such that it contains the most frequently used words or subwords. the original vocabulary released with BERT as BASEVOCAB. 对文本是通过BPE的方式将输入变为subword level。得到的字典包含most frequently userd words or subwords. 原始的BERT得到的字典被标记为BASEVOCAB。Use the SentencePiece1 library to build a new WordPiece vocabulary.  这里用了 [SentencePiece](<https://github.com/google/sentencepiece>)这个包。

**Corpus**： train SCIBERT on a random sample of 1.14M papers from Semantic Scholar. 18%
papers from the computer science domain and 82% from the broad biomedical domain. 用了全文进行训练，而不是仅仅abstract。通过[Science Parse](<https://github.com/allenai/science-parse>)进行pdf的解析。To split the documents into sentences, we use [ScispaCy](<https://github.com/allenai/SciSpaCy>). 

**模型图：**

**结果**：

用于进行测试的任务有：NER (Named Entity Recognition)， PICO (Participant Intervention Comparison Outcome Extraction)， CLS (Classification)， REL (Relation Classification)， DEP (Dependency Parsing) 。 

![image-20190404093400413](https://ws2.sinaimg.cn/large/006tKfTcgy1g1qavp4km6j30m00ck779.jpg)

The BERT pretrained models are converted to be compatible with PyTorch using the [pytorch pretrained-BERT library](<https://github.com/huggingface/pytorch-pretrained-bert>)

![image-20190404093650466](https://ws3.sinaimg.cn/large/006tKfTcgy1g1qaylzq6yj30o30jydkq.jpg)

不打分结果都是macro F1(span-level for NER, sentencelevel for REL and CLS, and token-level for PICO)。

**Effect of Vocabulary**：在新的domain上重新训练BERT embeddings,是有提升效果的。毕竟BASEVOCAB and SCIVOCAB有42%的重合。

**Effect of Casing**: we find for SCIBERT with SCIVOCAB that the cased model performs better than the uncased one on sequence tagging and parsing (+0.04 F1) and worse on sentence classification (-0.18 F1)。 Interestingly, BERT-Base and SCIBERT with BASEVOCAB both show better performance with uncased vocabularies on all tasks。可能在特定domain上训练出来的结果，用cased模型效果会更好吧。而general corpora的话则是uncased模型更好。

**BIOBERT**：比较了#78这篇基于biomedical的模型。效果上发现SCIBERT更好。
